### PR TITLE
Revert doctrine/inflector removal

### PR DESCRIPTION
**Pull request by samsonasik**
@TomasVotruba @staabm this is revert doctrine/inflector removal in case of too many BC changes.

### CODE REVIEW

**Comment by samsonasik** [comment]
packages tests error seems due to rector-src sync without update on rector-symfony

https://github.com/rectorphp/rector-src/actions/runs/6639191232/job/18037035712?pr=5201

so merge first is the way to go

**Comment by samsonasik** [comment]
@TomasVotruba Ready to merge 👍 , rector-symfony test will sync after merge

**Comment by TomasVotruba** [comment]
Seems like a way to go, as we'd have to take over whole domain of language for no gain.

Thank you for POC @staabm to verify this :+1: 

<br>

@samsonasik Feel free to merge when ready :+1:

**Comment by staabm** [comment]
:+1:

**Comment by samsonasik** [comment]
@TomasVotruba I don't have merge access as the rector-symfony still not green, that will require it to be merged first.

@TomasVotruba could you merge it as I don't have merge access ? Thank you.

**Comment by samsonasik** [comment]
@TomasVotruba I need your hand to merge it as dependency required by rector-symfony

File: .github/workflows/packages_tests.yaml [hunk 1]
```diff
@@ -56,5 +56,7 @@ jobs:
                 if: ${{ github.event_name == 'pull_request' }}
 
             -   run: composer install --ansi
+            # temporary to give me access to merge
+            -   run: composer require doctrine/inflector:^2.0.6
```
Comment (samsonasik) [inline .github/workflows/packages_tests.yaml:60]: @TomasVotruba this is temporary to give me merge access as it used by rector-symfony

Comment (samsonasik) [reply .github/workflows/packages_tests.yaml:60]: cleaned tweak on https://github.com/rectorphp/rector-src/pull/5202

**Pull request merged**